packages/edit-site/src/components/global-styles/font-library-modal/fonts-grid.js

```diff
@@ -29,30 +25,26 @@ function FontsGrid( { title, children, pageSize = 32 } ) {
 	}, [ lastItem ] );
 
 	return (
-		<div className="font-library-modal__fonts-grid">
-			<VStack spacing={ 0 }>
-				{ title && (
-					<>
-						<Text className="font-library-modal__subtitle">
-							{ title }
-						</Text>
-						<Spacer margin={ 2 } />
-					</>
-				) }
-				<div className="font-library-modal__fonts-grid__main">
-					{ items.map( ( child, i ) => {
-						if ( i === itemsLimit - 1 ) {
-							return (
-								<div key={ child.key } ref={ setLastItem }>
-									{ child }
-								</div>
-							);
-						}
-						return <div key={ child.key }>{ child }</div>;
-					} ) }
-				</div>
-			</VStack>
-		</div>
+		<VStack className="font-library-modal__fonts-grid" as="fieldset">
+			{ title && (
+				<legend className="font-library-modal__fonts-grid-legend">
+					<h2 className="font-library-modal__fonts-grid-title">
+						{ title }
+					</h2>
+				</legend>
+			) }
+			<ul className="font-library-modal__fonts-grid__list">
```

Comment:
Good idea using a list. This will help assistive technology to _count and announce_ the amount of items. Reminder that Safari and VoiceOver do not announce styled lists as list. The workaround is to add a `role="list"` even though that's technically redundant.

Reply:
Which of the following do you think is ideal for this context?

```html
	<ul>
		<li><button>...</button></li>
		<li><button>...</button></li>
	</ul>
```

```html
	<ul role="list">
		<li role="listitem"><button>...</button></li>
		<li role="listitem"><button>...</button></li>
	</ul>
```

```html
	<div class="list">
		<div class="listitem"><button>...</button></div>
		<div class="listitem"><button>...</button></div>
	</div>
```

Reply:
role="listitem" on the items isn't necessary.
It's Safari that doesn't expose a `<ul>` as a role `list` when it's styled. Long time ago I had a twitters thread with folks from the Safari team and they called this a `feature`. Actually, it's a non-standard and non-compliant implementation. In other words, it's a Safari bug.

Screenshots from the Safari dev tools, without and with role=llist:

![safari no ul role](https://github.com/WordPress/gutenberg/assets/1682452/5f309f1f-c5bf-457a-8412-79a037805942)

![safari ul role list](https://github.com/WordPress/gutenberg/assets/1682452/8bef141e-8143-41dc-bb04-25222fe4c40f)

Reply:
Fixed with [3f5630d](https://github.com/WordPress/gutenberg/pull/58834/commits/3f5630db0a9c5081ee123e90622c3b748a2e7089)

Thank you for explaining in detail. I looked over the Gutenberg project again and found that `role=list` was indeed assigned to the `ul` element. However, this attribute is detected by eslint as an error, so it seems like we need to use `eslint-disable/enable`.

Reply:
Oh yes it could also be reported as warning by some automated accessibility checkers. Technically not a real error, it's just re-stating the native role of a `<ul>`, which is technically redundant.

Reply:
I was totally unaware of this Safari bug and thanks for sharing @afercia